### PR TITLE
applications: machine_learning: Adapt to nRF54H20 with sysbuild

### PR DESCRIPTION
**Pull request by maje-emb**
Adapts the application to the nRF54H20 SoC and sysbuild.

Ref: NCSDK-27407

### CODE REVIEW

File: applications/machine_learning/configuration/nrf54h20dk_nrf54h20_cpuapp/sensor_manager_def.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	.cfg = {
+		.type = SENSOR_TRIG_MOTION,
+		.chan = SENSOR_CHAN_ACCEL_XYZ,
+	},
+	.activation = {
+		.type = ACT_TYPE_ABS,
+		.thresh = FLOAT_TO_SENSOR_VALUE(CONFIG_ADXL362_ACTIVITY_THRESHOLD *
+			(IS_ENABLED(CONFIG_ADXL362_ACCEL_RANGE_8G) ? 8.0 :
+			 IS_ENABLED(CONFIG_ADXL362_ACCEL_RANGE_4G) ? 4.0 :
+			 IS_ENABLED(CONFIG_ADXL362_ACCEL_RANGE_2G) ? 2.0 : 1/0)
```
Comment (dchat-nordic) [inline applications/machine_learning/configuration/nrf54h20dk_nrf54h20_cpuapp/sensor_manager_def.h:28]: Is the division here correct?

File: applications/machine_learning/remote/Kconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menu "Machine learning application (remote core)"
+
+rsource "../common/Kconfig"
+# Kconfig.ml_app should land in "../common/src/" but im not
+# placing it there just to avoid conflicts when HCS will is synced with NCS
```
Comment (dchat-nordic) [inline applications/machine_learning/remote/Kconfig:11]: This should me rephrased.

**Review by dchat-nordic — CHANGES_REQUESTED** [review]

File: applications/machine_learning/app_desc.rst [hunk 3]
```diff
@@ -355,6 +359,32 @@ Building the default configurations requires an Internet connection, because the
...
+
+   or use twister test case:
+
+   .. code-block:: console
+
+       west build -b nrf54h20dk/nrf54h20/cpuapp -T applications.machine_learning.sensor_hub.zdebug.singlecore .
+
+   To build the application for the nRF54H20 PDK with the simulated sensor on the PPR core (dual-core application), run the following command:
+
+      .. code-block:: console
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:378]:
```suggestion
   To build the application for the nRF54H20 PDK with the simulated sensor on the PPR core (dual-core application), run the following command:

   .. code-block:: console
```

File: applications/machine_learning/common/Kconfig [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (peknis) [inline applications/machine_learning/common/Kconfig:2]:
```suggestion
# Copyright (c) 2024 Nordic Semiconductor ASA
```

Comment (peknis) [reply applications/machine_learning/common/Kconfig:2]: Update in other new Kconfig and header files as well.

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by grochu — APPROVED** [review]
Looks good to me.

File: applications/machine_learning/configuration/nrf54h20dk_nrf54h20_cpuapp/sensor_manager_def.h (hunk 1 above)
Comment (maje-emb) [reply applications/machine_learning/configuration/nrf54h20dk_nrf54h20_cpuapp/sensor_manager_def.h]: It looks like it is not correct. Corrected.

**Comment by zycz** [comment]
aggregator and proxy looks good

**Review by zycz — APPROVED** [review]

File: applications/machine_learning/app_desc.rst [hunk 5]
```diff
@@ -355,6 +360,32 @@ Building the default configurations requires an Internet connection, because the
...
+
+      west build -b nrf54h20dk/nrf54h20/cpuapp -- -DSHIELD=pca63566 -DFILE_SUFFIX="singlecore"
+
+   or use twister test case:
+
+   .. code-block:: console
+
+       west build -b nrf54h20dk/nrf54h20/cpuapp -T applications.machine_learning.sensor_hub.zdebug.singlecore .
+
+   To build the application for the nRF54H20 PDK with the simulated sensor on the PPR core (dual-core application), run the following command:
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:377]:
```suggestion
   To build the application for the nRF54H20 DK with the simulated sensor on the PPR core (dual-core application), run the following command:
```
It is a DK, not a PDK, right?

**Review by peknis — CHANGES_REQUESTED** [review]

File: applications/machine_learning/app_desc.rst (hunk 5 above)
Comment (maje-emb) [reply applications/machine_learning/app_desc.rst]: Right.

**Review by peknis — APPROVED** [review]

File: applications/machine_learning/app_desc.rst [hunk 6]
```diff
@@ -353,7 +358,32 @@ Building and running
 The nRF Machine Learning application is built the same way as any other |NCS| application or sample.
 Building the default configurations requires an Internet connection, because the machine learning model source files are downloaded from the web during the application build.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:360]:
```suggestion

.. |application path| replace:: :file:`applications/machine_learning`

```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**